### PR TITLE
HITL - Handle task metrics.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset adds handling of task metrics:
* Add a thin wrapper around common metrics to access them safely.
* Use `task_percent_complete` for defining episode success.
* Log `task_percent_complete` and `task_explanation`.

## How Has This Been Tested

Tested on HITL applications.

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/rearrange_v2/metrics.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from habitat_hitl.app_states.app_service import AppService
+
+
+class Metrics:
```
Comment (eundersander) [inline examples/hitl/rearrange_v2/metrics.py:7]: IMO a plural word is a bad choice for a class name. E.g. what if one day you need a list of these, what would you call that variable? `metricses`?

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 2]
```diff
@@ -788,10 +804,22 @@ def _is_episode_finished(self) -> bool:
 
     def _is_episode_successful(self) -> bool:
         """
-        Returns true if all agents finished the episode successfully.
+        Returns true if:
+        * Task success is >99%.
```
Comment (eundersander) [inline examples/hitl/rearrange_v2/rearrange_v2.py:808]: Why >99%? Looks like a magic number. How about a comment.

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 3]
```diff
@@ -788,10 +804,22 @@ def _is_episode_finished(self) -> bool:
...
+        * All agents finished the episode without reporting an error.
         """
-        return all(
+
+        task_percent_complete = self._metrics.get_task_percent_complete()
+        task_successful = (
```
Comment (eundersander) [inline examples/hitl/rearrange_v2/rearrange_v2.py:813]: What is the real situation for `rearrange_v2` that this metric is None? How about add a comment explaining why sometimes that metric is missing and why we want to consider the task successful in that case.

**Review by eundersander — APPROVED** [review]
I left some comments otherwise LGTM!

File: examples/hitl/rearrange_v2/metrics.py (hunk 1 above)
Comment (0mdc) [reply examples/hitl/rearrange_v2/metrics.py:7]: We can rename the class.

I will do this separately as a refactor pass.

**Pull request merged**